packages/jni-swig-stub/src/main/jni/realm_api_helpers.cpp

```diff
@@ -34,6 +34,26 @@ jobject wrap_pointer(JNIEnv* jenv, jlong pointer, jboolean managed = false) {
                            managed);
 }
 
+inline jboolean jni_check_exception(JNIEnv *jenv = get_env(), bool registrable_callback_error = false) {
+    if (jenv->ExceptionCheck()) {
+        jthrowable exception = jenv->ExceptionOccurred();
+        jenv->ExceptionClear();
+        // setting the user code error is only propagated on certain callbacks.
+        if (registrable_callback_error)
+            realm_register_user_code_callback_error(jenv->NewGlobalRef(exception));
+        return false;
+    }
+    return true;
+}
+
+inline void push_local_frame(JNIEnv *jenv, jint frame_size) {
+    if (jenv->PushLocalFrame(frame_size) != 0) {
+        jni_check_exception(jenv);
+        jenv->ExceptionDescribe();
```

Comment:
The exception is already cleared in https://github.com/realm/realm-kotlin/pull/1560/files#diff-c71ff0791782da90014db6f7000bdeea6c883ca9729f17bd1b11041d1e7711faR40 so this won't print anything to stderr? 
I think instead of calling `jenv->ExceptionClear();` inside `jni_check_exception` we could call `jenv->ExceptionDescribe();` which has the side effect of also clearing the exception

Reply:
Yes, it is my bad. It should be inside.